client/src/main/java/org/spine3/base/MapStringifier.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.base;
+
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static com.google.common.collect.Maps.newHashMap;
+import static org.spine3.util.Exceptions.newIllegalArgumentException;
+
+/**
+ * The stringifier for the {@code Map} classes.
+ *
+ * <p>The stringifier for the type of the elements in the map
+ * should be registered in the {@code StringifierRegistry} class
+ * for the correct usage of {@code MapStringifier}.
+ *
+ * <h3>Example</h3>
+ *
+ * {@code
+ *  // The registration of the stringifier.
+ *  final Type type = new TypeToken<Map<String, Long>>(){}.getType();
+ *  StringifierRegistry.getInstance().register(stringifier, type);
+ *
+ *  // Obtain already registered `MapStringifier`.
+ *  final Stringifier<Map<String, Long>> mapStringifier = StringifierRegistry.getInstance()
+ *                                                                           .getStringifier(type);
+ *
+ *  // Convert to string.
+ *  final Map<String, Long> mapToConvert = ...
+ *  final String convertedString = mapStringifier.toString(mapToConvert);
+ *
+ *  // Convert from string.
+ *  final String stringToConvert = ...
+ *  final Map<String, Long> convertedMap = mapStringifier.fromString(stringToConvert);
+ * }
+ *
+ * @param <K> the type of the keys in the map
+ * @param <V> the type of the values in the map
+ */
+class MapStringifier<K, V> extends Stringifier<Map<K, V>> {
+
+    private static final char DEFAULT_ELEMENT_DELIMITER = ',';
+    private static final String ESCAPE_SEQUENCE = "\\";
+    private static final String KEY_VALUE_DELIMITER = ESCAPE_SEQUENCE + ':';
+
+    /**
+     * The delimiter for the passed elements in the {@code String} representation,
```

Comment:
That also should be mentioned in the `MapStringifier(Class<K> keyClass, Class<V> valueClass)` constructor documentation, which now is missing.